solutions/cf-1528A.mdx

```diff
@@ -0,0 +1,110 @@
+---
+id: cf-1528A
+source: CF
+title: Parsa's Humongous Tree
+author: Ryan Chou
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/91058)
+
+## Explanation
+
+Let's go through a case where we've already computed the beauty values for all of vertex $u$'s neighbors.
+
+Let $\texttt{range}_u$ store $l_u$ and $r_u$ and $\texttt{adj}_u$ store the beauty values of all vertices adjacent to $u$.
+
+$\texttt{range}_u = \{1, 10\}$
+
+$\texttt{adj}_u = \{2, 3, 6, 9\}$
+
+Let's say that we select $7$, can we do better?
+
+The current beauty sum is: $|7-2| + |7-3| + |7-6| + |7-9| = 12$.
+
+Can we do better?
+
+Intuitively, for any $X$ that we choose to be $u$'s beauty value, if there's more elements that are greater than this $X$, making $X$ as small as possible would be more optimal (increasing differences for these elements) and vice versa. Note that in a case where the number of greater beauty values is the same as the number of smaller beauty values, setting $X$ to the minimum or maximum will never decrease the answer.
```

Comment:
hey maybe you should include that thing about the median
you know how if you wanna minimize the error you select the median?
well in this case we wanna go as goddamn far from the median as possible
and then you can link to a math.stackexchange link or something

Reply:
sure